### PR TITLE
2034-newAdjestUrnRegex

### PR DESCRIPTION
**Pull request by TobiasNx**
Replaces #235

This PR tries to catch all URN resolver links not just nbn from 865.

It uses the URNs from 024 as fallback if no 865 links with URN are provided.
Also it changes the nbn links to org when creating them as fallback.

### CODE REVIEW

**Review by maipet — APPROVED** [review]

File: src/main/resources/alma/fix/identifiers.fix [hunk 1]
```diff
@@ -17,15 +17,18 @@ end
 # Sometimes urn are not set in 024 then we could pick up the missing from 856.
 # 856 - Electronic Location and Access (R) - Subfield: $u (R) $3 (NR)
 # 1. Indicator: 4 = HTTP
+set_array("@urnLinks")
+
 do list(path:"8564?", "var":"$i")
-  if all_match("$i.u", "^http[s]?:\\/\\/nbn-resolving.(de|org)\\/(.*)")
+  if all_match("$i.u", "^http.*(urn=|[org|de]/)(urn:.+$)") # This should ignore repository links like: https://sammlungen.ulb.uni-muenster.de/urn/urn:nbn:de:hbz:6-85659520092
```
[the remark this answers is not in the document]
Comment (TobiasNx) [reply src/main/resources/alma/fix/identifiers.fix:23]: like this ?

```suggestion
  if all_match("$i.u", "^http.*(urn=|\\.(org|de)/)urn:.+$") # This should ignore repository links like: https://sammlungen.ulb.uni-muenster.de/urn/urn:nbn:de:hbz:6-85659520092
```

[the remark this answers is not in the document]
Comment (TobiasNx) [reply src/main/resources/alma/fix/identifiers.fix]: Done  thanks!

**Comment by TobiasNx** [comment]
@blackwinter since you became a reviewer by commenting, your approval is still pending.

**Comment by blackwinter** [comment]
But it's not required, is it? I'm just offering suggestions. Feel free to deal with them however you see fit.

If you're asking for approval, though, I'm inclinded to _suggest_ that you adjust the other regexp as well.

**Comment by TobiasNx** [comment]
> But it's not required, is it? I'm just offering suggestions. Feel free to deal with them however you see fit.
> 
> If you're asking for approval, though, I'm inclinded to _suggest_ that you adjust the other regexp as well.

Feel free to suggest the change for the other one, always good to improve code.

File: src/main/resources/alma/fix/identifiers.fix [hunk 2]
```diff
@@ -17,15 +17,18 @@ end
...
 # 1. Indicator: 4 = HTTP
+set_array("@urnLinks")
+
 do list(path:"8564?", "var":"$i")
-  if all_match("$i.u", "^http[s]?:\\/\\/nbn-resolving.(de|org)\\/(.*)")
+  if all_match("$i.u", "^http.*(urn=|\\.(org|de)/)urn:.+$") # This should ignore repository links like: https://sammlungen.ulb.uni-muenster.de/urn/urn:nbn:de:hbz:6-85659520092
     copy_field("$i.u", "urn[].$append")
-    replace_all("urn[].$last", "^http[s]?:\\/\\/nbn-resolving.(de|org)\\/(.*)", "$2")
+    copy_field("$i.u", "@urnLinks.$append")
+    replace_all("urn[].$last", "^http.*[/=](urn:.+$)", "$1")
```
Comment (blackwinter) [inline src/main/resources/alma/fix/identifiers.fix:26]:
```suggestion
    replace_all("urn[].$last", "^http.*[/=]urn:.+$", "$1")
```

File: src/main/resources/alma/fix/relatedRessourcesAndLinks.fix [hunk 3]
```diff
@@ -271,7 +271,7 @@ set_array("fulltextOnline[]")
 
 do list(path: "8564?", "var":"$i")
   if exists("$i.u")
-    unless any_match("$i.u",".*(doi.org|nbn-resolving).*")
+    unless any_match("$i.u",".*(doi.org|urn=urn:|(org|de)/urn:).*") # This should not skip repository links like: https://sammlungen.ulb.uni-muenster.de/urn/urn:nbn:de:hbz:6-85659520092
```
Comment (blackwinter) [inline src/main/resources/alma/fix/relatedRessourcesAndLinks.fix:274]:
```suggestion
    unless any_match("$i.u",".*(doi.org|urn=urn:|\\.(org|de)/urn:).*") # This should not skip repository links like: https://sammlungen.ulb.uni-muenster.de/urn/urn:nbn:de:hbz:6-85659520092
```

**Review by blackwinter — COMMENTED** [review]
If we wanted to be even more pedantic, we'd change all the capturing parentheses (`(...)`) to non-capturing ones (`(?:...)`).

**Pull request merged**